src/ros_filter.cpp

```diff
@@ -844,11 +844,11 @@ void RosFilter<T>::loadParams()
 
   // Determine if we're using a control term
   double control_timeout = sensor_timeout;
-  std::vector<bool> control_update_vector(TWIST_SIZE, 0);
-  std::vector<double> acceleration_limits(TWIST_SIZE, 1.0);
-  std::vector<double> acceleration_gains(TWIST_SIZE, 1.0);
-  std::vector<double> deceleration_limits(TWIST_SIZE, 1.0);
-  std::vector<double> deceleration_gains(TWIST_SIZE, 1.0);
+  std::vector<bool> control_update_vector;
+  std::vector<double> acceleration_limits;
```

Comment:
So re-reading this, I feel like this changes the filter's behaviour (at least the ROS1 behaviour).

Before, if the user failed to specify, e.g., `acceleration_limits`, then we would just use the default values that were already here (a vector of six 1.0 values). [Now we just have an empty vector](https://github.com/cra-ros-pkg/robot_localization/blob/b70d03157c092871308b95c8d7de091521f2dfd5/src/ros_filter.cpp#L884), no? I'm not familiar with ROS2's parameter declaration scheme, but I'm guessing that if the user turns on `use_control`, but fails to specify any limits, these vectors will all be empty. I feel like the `else` clauses for failed parameter loading need to initialize these, e.g., you need to add this [here](https://github.com/cra-ros-pkg/robot_localization/blob/b70d03157c092871308b95c8d7de091521f2dfd5/src/ros_filter.cpp#L884):

```
acceleration_limits.resize(TWIST_SIZE, 1.0);
```

...but again, this all depends on whether `get_parameter` returns true, even if the user didn't specify the parameter.

Reply:
But CI still failed..

Reply:
Oh sorry, I wasn't suggesting that this was causing the CI failure. I was just pointing it out as a change in behaviour.

Are you able to run these tests locally? And can you run them with and without your changes?

Reply:
@ayrton04 
I could use colcon_test to locally test its and finally solved the problem.
Since `control_update_vector` will be used when it was not initialized, the tests were not passed. I add resize to vector in `else` statement that will be activated when `use_control` param was set false.